mixer/tools/mixgen/cmd/api.go

```diff
@@ -12,27 +12,76 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package interfacegen
+package cmd
 
 import (
 	"bytes"
 	"fmt"
 	"go/format"
 	"io/ioutil"
 	"os"
+	"path/filepath"
 	"regexp"
 	"strings"
 	"text/template"
 
 	"github.com/gogo/protobuf/proto"
 	"github.com/gogo/protobuf/protoc-gen-gogo/descriptor"
+	"github.com/spf13/cobra"
 	"golang.org/x/tools/imports"
 
+	"istio.io/istio/mixer/cmd/shared"
 	descriptor2 "istio.io/istio/mixer/pkg/protobuf/descriptor"
-	tmpl "istio.io/istio/mixer/tools/codegen/pkg/interfacegen/template"
 	"istio.io/istio/mixer/tools/codegen/pkg/modelgen"
 )
 
+func apiGenCmd(rawArgs []string, printf, fatalf shared.FormatFn) *cobra.Command {
+	var templateFile string
+	var outInterfaceFile string
+	var oAugmentedTmplFile string
+	var mappings []string
+	adapterCmd := &cobra.Command{
+		Use:   "api",
+		Short: "creates type and service definitions from a given template",
+		Run: func(cmd *cobra.Command, args []string) {
+			var err error
+
+			outInterfaceFile, err = filepath.Abs(outInterfaceFile)
+			if err != nil {
+				fatalf("Invalid path %s: %v", outInterfaceFile, err)
+			}
+			oAugmentedTmplFile, err = filepath.Abs(oAugmentedTmplFile)
+			if err != nil {
+				fatalf("Invalid path %s: %v", oAugmentedTmplFile, err)
+			}
+			importMapping := make(map[string]string)
+			for _, maps := range mappings {
+				m := strings.Split(maps, ":")
+				importMapping[strings.TrimSpace(m[0])] = strings.TrimSpace(m[1])
+			}
+
+			generator := Generator{OutInterfacePath: outInterfaceFile, OAugmentedTmplPath: oAugmentedTmplFile, ImptMap: importMapping}
+			if err := generator.Generate(templateFile); err != nil {
+				fatalf("%v", err)
+			}
+		},
+	}
+	adapterCmd.PersistentFlags().StringVarP(&templateFile, "template", "t", "", "Input "+
+		"template file path")
+
+	adapterCmd.PersistentFlags().StringVar(&outInterfaceFile, "out_go", "./generated.go", "Output "+
```

Comment:
it might be nice to match the protoc plugin style here. so, something like `go_out` and `template_out`.